src/ECER.Resources.Documents/Applications/ApplicationRepository.cs

```diff
@@ -63,10 +65,46 @@ public async Task<string> SaveDraft(Application application)
       context.UpdateObject(ecerApplication);
     }
     _ = UpdateApplicationTranscripts(ecerApplication, ecerTranscripts);
+    _ = UpdateApplicationWorkExperienceReferences(ecerApplication, ecerWorkExperienceReferences);
     context.SaveChanges();
     return ecerApplication.ecer_ApplicationId.Value.ToString();
   }
 
+  public async Task UpdateApplicationWorkExperienceReferences(ecer_Application application, List<ecer_WorkExperienceRef> updatedReferences)
+  {
+    await Task.CompletedTask;
+    var existingWorkExpericnes = context.ecer_WorkExperienceRefSet.Where(t => t.ecer_Applicationid.Id == application.Id).ToList();
+
+    // 1. Remove WorkExperienceReferences that they exist in the dataverse but not in the application
+    foreach (var reference in existingWorkExpericnes)
+    {
+      if (!updatedReferences.Any(t => t.Id == reference.Id))
+      {
+        context.DeleteObject(reference);
```

Comment:
@farzadnadiri can we verify with the Dataverse folks if they want us to delete or change status please?

Reply:
I reached Dataverse folks and asked my question but haven't heard back yet. I need to merge this because we need this merged to start working on the FE task. If we decide to change it, will fix it in the story branch.